### PR TITLE
Tweaks to datalog Attribute API to reduce verbosity.

### PR DESCRIPTION
**Pull request by bgogul**
This PR does some minor tweaks to reduce verbosity when defining attributes. This also adds a unit test.

### CODE REVIEW

File: src/ir/datalog/attribute.h [hunk 1]
```diff
@@ -24,34 +24,41 @@
...
-  StringAttributePayload(Symbol symbol)
-      : AttributePayload(kStringAttributePayloadName) {
-    arguments_.push_back(std::make_unique<Symbol>(symbol));
-  }
-};
+  class String : public AttributePayload {
+   public:
+    explicit String(Symbol symbol)
+        : AttributePayload(kStringAttributePayloadName) {
+      arguments_.push_back(std::make_unique<Symbol>(symbol));
```
Comment (markww) [inline src/ir/datalog/attribute.h:39]: Should probably `std::move(symbol)` here.

**Review by markww — APPROVED** [review]
LGTM. Minor nit, opportunity to move in a ctor not taken, good otherwise.

**Review by harsha-mandadi — APPROVED** [review]
LGTM.

File: src/ir/datalog/attribute.h (hunk 1 above)
Comment (bgogul) [reply src/ir/datalog/attribute.h]: done.

**Pull request closed without merging**